src/Integration/LocalServices/UnboundProjectFinder.cs

```diff
@@ -64,14 +72,20 @@ private IEnumerable<Project> GetUnboundProjects(BindingConfiguration binding)
             var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
             projectSystem.AssertLocalServiceIsNotNull();
 
-            var filteredSolutionProjects = projectSystem.GetFilteredSolutionProjects();
+            var filteredSolutionProjects = projectSystem.GetFilteredSolutionProjects().ToArray();
+            logger.LogDebug($"[Binding check] Number of bindable projects: {filteredSolutionProjects.Length}");
+
 
             return filteredSolutionProjects
                 .Where(project =>
                 {
                     var configProjectBinder = projectBinderFactory.Get(project);
 
-                    return configProjectBinder.IsBindingRequired(binding, project);
+                    logger.LogDebug($"[Binding check] Checking binding for project '{project.Name}'. Binder type: {configProjectBinder.GetType().Name}");
+                    var required = configProjectBinder.IsBindingRequired(binding, project);
+                    logger.LogDebug($"[Binding check] Is binding required: {required} (project: {project.Name})");
```

Comment:
Would project.Name work for open as folder projects? I can't remember. I remember some specific properties returning empty strings.

Reply:
Good point. I'm not sure what we could return instead in that case.
I'll try it and see.

Reply:
Actually, "open as folder" projects should have been filtered out by this point since we shouldn't recognise them as being bind-able. I'll check, though.

Reply:
Update: you can't mix "real" projects and "open as folder" projects in a solution.
"Open as folder" projects are not bindable in the first place, so we'd never get to this point.

Reply:
What would happen in CMake's case? Would we not support connected mode for CMake?

Reply:
At the moment we don't detect it as a bindable project type. We'd have to tackle that as part of the sprint.